index_classes.py

```diff
@@ -0,0 +1,108 @@
+        headers={"x-cosmos-block-height": str(height)},
+    )
+    resp.raise_for_status()
+    return resp.json()["issuers"]
+
+
+def _index_classes(pg_conn, _client, _chain_num):
+    with pg_conn.cursor() as cur:
+        for event in events_to_process(
+            cur,
+            "credit_classes",
+        ):
+            (
+                type,
+                block_height,
+                tx_idx,
+                msg_idx,
+                _,
+                _,
+                chain_num,
+                timestamp,
+                tx_hash,
+            ) = event[0]
+            normalize = {}
+            normalize["type"] = type
+            normalize["block_height"] = block_height
+            normalize["tx_idx"] = tx_idx
+            normalize["msg_idx"] = msg_idx
+            normalize["chain_num"] = chain_num
+            normalize["timestamp"] = timestamp
+            normalize["tx_hash"] = tx_hash
+            for entry in event:
+                (_, _, _, _, key, value, _, _, _) = entry
+                value = value.strip('"')
+                normalize[key] = value
+            logger.info(normalize)
+            issuers = fetch_issuers(normalize["block_height"], normalize["class_id"])
+            logger.info(normalize)
+            logger.info(issuers)
+            insert_text = textwrap.dedent(
+                """
+            INSERT INTO credit_classes (
+                type,
+                block_height,
+                tx_idx,
+                msg_idx,
+                chain_num,
+                timestamp,
+                tx_hash,
+                credit_class_id,
+                issuer_address
+            ) VALUES (
+                %s,
+                %s,
+                %s,
+                %s,
+                %s,
+                %s,
+                %s,
+                %s,
+                %s
+            );"""
+            ).strip("\n")
+            with pg_conn.cursor() as _cur:
+                _cur.execute(
+                    "UPDATE credit_classes SET latest='f' WHERE credit_class_id = %s;",
+                    (normalize["class_id"],),
+                )
+                logger.info(_cur.statusmessage)
+                for issuer in issuers:
+                    row = (
+                        normalize["type"],
+                        normalize["block_height"],
+                        normalize["tx_idx"],
+                        normalize["msg_idx"],
+                        normalize["chain_num"],
+                        normalize["timestamp"],
+                        normalize["tx_hash"],
+                        normalize["class_id"],
+                        issuer,
+                    )
+                    _cur.execute(
+                        insert_text,
+                        row,
+                    )
+                    logger.info(_cur.statusmessage)
+                pg_conn.commit()
+                logger.info("credit class issuers updated..")
```

Comment:
Only one issuer is being added here.

```suggestion
                logger.info("credit class issuer added..")
```

Reply:
plural makes sense, we commit after all inserts are taken care of